media/js/base/site.js

```diff
@@ -146,12 +146,17 @@
         // to avoid lots of flickering
         var platform = window.site.platform = window.site.getPlatform();
         var version = window.site.platformVersion = window.site.getPlatformVersion();
+        var _version = version ? parseFloat(version) : 0;
 
         if (platform === 'windows') {
             // Add class to support downloading Firefox for Windows 64-bit on Windows 7 and later
-            if (version && parseFloat(version) >= 6.1) {
+            if (_version >= 6.1) {
                 h.className += ' win7up';
             }
+            // Add class to support downloading Firefox 52 ESR on Windows XP/Vista
+            if (_version >= 5.1 && _version <= 6) {
+                h.className += ' xpvista';
```

Comment:
Some unit tests here would be good.

Reply:
IIRC we have never had unit tests for the class names on `<body>` but I agree to have some tests. So I’m making a change to `getPlatform()` in [site.js](https://github.com/mozilla/bedrock/blob/master/media/js/base/site.js) and update the relevant test accordingly. Basically Windows XP and Vista should be `oldwin`.

Reply:
Updated `getPlatform()` to detect XP/Vista as `oldwin` while using `windows xpvista` as the class name on `<body>` to continue showing the download button on these legacy platforms.